waffle_hub/dataset/dataset.py

```diff
@@ -731,29 +731,34 @@ def from_coco(
         """
         ds = Dataset.new(name=name, task=task, root_dir=root_dir)
 
-        if isinstance(coco_file, list) and isinstance(coco_root_dir, list):
-            if len(coco_file) != len(coco_root_dir):
-                raise ValueError("coco_file and coco_root_dir should have same length.")
-        if not isinstance(coco_file, list) and isinstance(coco_root_dir, list):
-            raise ValueError(
-                "ambiguous input. The number of coco_file should be same or greater than coco_root_dir."
-            )
-        if not isinstance(coco_file, list):
-            coco_file = [coco_file]
-        if not isinstance(coco_root_dir, list):
-            coco_root_dir = [coco_root_dir] * len(coco_file)
+        try:
+            if isinstance(coco_file, list) and isinstance(coco_root_dir, list):
+                if len(coco_file) != len(coco_root_dir):
+                    raise ValueError("coco_file and coco_root_dir should have same length.")
+            if not isinstance(coco_file, list) and isinstance(coco_root_dir, list):
+                raise ValueError(
+                    "ambiguous input. The number of coco_file should be same or greater than coco_root_dir."
+                )
+            if not isinstance(coco_file, list):
+                coco_file = [coco_file]
+            if not isinstance(coco_root_dir, list):
+                coco_root_dir = [coco_root_dir] * len(coco_file)
 
-        coco_files = coco_file
-        coco_root_dirs = coco_root_dir
+            coco_files = coco_file
+            coco_root_dirs = coco_root_dir
 
-        import_coco(ds, coco_files, coco_root_dirs)
+            import_coco(ds, coco_files, coco_root_dirs)
 
-        if len(coco_files) == 2:
-            logger.info("copying val set to test set")
-            io.copy_file(ds.val_set_file, ds.test_set_file, create_directory=True)
+            if len(coco_files) == 2:
+                logger.info("copying val set to test set")
+                io.copy_file(ds.val_set_file, ds.test_set_file, create_directory=True)
```

Comment:
In the explanation, it states, "do not export test set if test_ratio was set to 0.".  However, it seems that here, val and test set is same.